spec/models/collection_spec.rb

```diff
@@ -120,9 +120,89 @@ class Member < ActiveFedora::Base
+        allow(Hyrax::CollectionType).to receive(:find_by_gid!).with('gid://internal/hyrax-collectiontype/5').and_return(collection_type)
+        allow(collection_type).to receive(:persisted?).and_return(true)
+      end
+
+      it 'loads instance of collection type based on gid' do
+        collection.collection_type_gid = 'gid://internal/hyrax-collectiontype/5'
+        collection.save!
+        col = described_class.find(collection.id)
+        expect(col.collection_type).to be_a Hyrax::CollectionType
+        expect(col.collection_type.id).to eq 5
+      end
+    end
+
+    context 'when gid in collection object is nil' do
+      let(:collection) { described_class.new(title: ['title']) }
+
+      before do
+        allow(collection).to receive(:collection_type_gid).and_return(nil)
+      end
+
+      it 'loads default collection type' do
+        collection.save!
+        col = described_class.find(collection.id)
+        expect(col.collection_type).to be_a Hyrax::CollectionType
+        expect(col.collection_type.machine_id).to eq Hyrax::CollectionType::DEFAULT_ID
+      end
+    end
+  end
+
+  describe '#collection_type_gid=' do
+    let(:collection) { described_class.new }
+
+    it 'sets gid' do
+      gid = 'gid://internal/hyrax-collectiontype/10'
+      collection.collection_type_gid = gid
+      expect(collection.collection_type_gid).to eq gid
+    end
+  end
+
+  describe '#collection_type' do
+    let(:collection) { described_class.new }
+
+    it 'returns nil if gid is nil' do
+      collection.collection_type_gid = nil
+      expect(collection.collection_type).to be_nil
+    end
+
+    it 'returns collection_type if already set' do
+      gid89 = 'gid://internal/hyrax-collectiontype/89'
+      allow(Hyrax::CollectionType).to receive(:find_by_gid!).with(gid89).and_return(Hyrax::CollectionType.new(id: 89))
+      collection.collection_type_gid = gid89
+      expect(collection.collection_type).to be_kind_of(Hyrax::CollectionType)
+      expect(collection.collection_type.gid).to eq gid89
+    end
+
+    it 'will not change value' do
+      gid89 = 'gid://internal/hyrax-collectiontype/89'
+      gid99 = 'gid://internal/hyrax-collectiontype/99'
+      allow(Hyrax::CollectionType).to receive(:find_by_gid!).with(gid89).and_return(Hyrax::CollectionType.new(id: 89))
+      collection.collection_type_gid = gid89
+      collection.collection_type
+      collection.collection_type_gid = gid99
+      expect(collection.collection_type).to be_kind_of(Hyrax::CollectionType)
+      expect(collection.collection_type.gid).to eq gid89
+    end
+
+    it 'throws StandardError if cannot find collection type for the gid' do
+      gid = 'gid://internal/hyrax-collectiontype/999'
+      collection.collection_type_gid = gid
+      # TODO: Should we capture the ActiveRecord error and produce something nicer?
+      expect { collection.collection_type }.to raise_error(StandardError, "Couldn't find Hyrax::CollectionType matching GID '#{gid}'")
```

Comment:
I believe this should be an `ActiveRecord::RecordNotFound` error (which is a descendant of `StandardError`)